### PR TITLE
Search bar implemented

### PR DESCRIPTION
**Pull request by lukan98**
Video demonstration:
https://user-images.githubusercontent.com/53893982/126450791-c049740d-f353-421e-9f67-5c1834be5bd0.mp4

### CODE REVIEW

File: MovieApp/PresentationLayer/Components/BaseSearchField/BaseSearchField.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import UIKit
+
+class BaseSearchField: UITextField {
+
+    static let defaultHeight = 40
+
+    override func leftViewRect(forBounds bounds: CGRect) -> CGRect {
+        CGRect(x: 0, y: 0, width: BaseSearchField.defaultHeight, height: BaseSearchField.defaultHeight)
```
Comment (istaudacher) [inline MovieApp/PresentationLayer/Components/BaseSearchField/BaseSearchField.swift]: You can define `static let defaultSize = CGSize(width: 40, height: 40)` and then have here `CGRect(origin: .zero, size: defaultSize)` and `CGRect(origin: CGPoint(Int(bounds.maxX) - dimension, y: 0), size: defaultSize)` below.

File: MovieApp/PresentationLayer/Components/SearchBar/SearchBar+Design.swift [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    func styleViews() {
+        searchField.backgroundColor = UIColor(named: "Gray")
+        searchField.layer.cornerRadius = 10
+        searchField.leftViewMode = .always
+        searchField.rightViewMode = .always
+
+        let placeholder = NSAttributedString(
+            string: "Search",
+            attributes: [
```
Comment (istaudacher) [inline MovieApp/PresentationLayer/Components/SearchBar/SearchBar+Design.swift:43]: To make this array more clear, extract fonts into variables. Also, do you need to cast to Any?

File: MovieApp/PresentationLayer/HomeScreen/HomeScreenViewController+Design.swift [hunk 3]
```diff
@@ -23,6 +30,13 @@ extension HomeScreenViewController: ConstructViewsProtocol {
             $0.top.leading.trailing.equalToSuperview()
             $0.height.equalTo(NavBarView.defaultHeight)
         }
+
+        searchBar.snp.makeConstraints {
+            $0.top.equalTo(navigationView.snp.bottom).offset(20)
+            $0.leading.equalToSuperview().offset(20)
```
Comment (istaudacher) [inline MovieApp/PresentationLayer/HomeScreen/HomeScreenViewController+Design.swift]: Try `$0.leading.trailing.equalToSuperview().inset(20)`

**Review by istaudacher — CHANGES_REQUESTED** [review]

File: MovieApp/PresentationLayer/Components/SearchBar/SearchBar+Design.swift (hunk 2 above)
Comment (lukan98) [reply MovieApp/PresentationLayer/Components/SearchBar/SearchBar+Design.swift:43]: The problem is that the font is an optional when constructed like this.
If I don't cast to Any XCode gives me a warning saying: Expression implicitly coerced from 'UIFont?' to 'Any'.
I have the option to provide a default value, force unwrap or cast to Any.

**Review by istaudacher — APPROVED** [review]

**Pull request merged**